[CARBONDATA-1846] Incorrect output on presto CLI while executing IN operator with multiple load

anubhav100 opened the pull request:
https://issues.apache.org/jira/browse/CARBONDATA-1846

Problem: problem is that in operator on timestamp is not working correctly because timestamp was getting multiplied by 1000 twice when we are creating inexpression

CarbonDataQA commented:
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1912/

CarbonDataQA commented:
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/684/

ravipesala commented:
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2245/

anubhav100 commented:
retest this please

CarbonDataQA commented:
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/687/

CarbonDataQA commented:
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1916/

anubhav100 commented:
@chenliang can you please review this PR?

CarbonDataQA commented:
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/946/

CarbonDataQA commented:
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2175/

anubhav100 commented:
retest this please

chenliang613 commented:
retest this please

chenliang613 commented:
@anubhav100  can you add one test case for this issue under presto module?

anubhav100 commented:
@chenliang613 yes i am adding it

anubhav100 commented:
@chenliang613 i added a test case for in operator on timestamp data type

CarbonDataQA commented:
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2255/

anubhav100 commented:
retest this please

CarbonDataQA commented:
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1032/

ravipesala commented:
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2511/

anubhav100 commented:
retest sdv please

CarbonDataQA commented:
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1035/

CarbonDataQA commented:
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2258/

CarbonDataQA commented:
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2263/

CarbonDataQA commented:
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1041/

anubhav100 commented:
retest this please

CarbonDataQA commented:
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1064/

chenliang613 commented:
@anubhav100  One small question : as per the jira description, need use multiple load to reproduce this issue, but your test case doesn't do multiple load.

anubhav100 commented:
@chenliang613  1.jira title is not completely correct it just appears to the tester that it is because of multiple load only problem is that in operator on timestamp is not working correctly because timestamp was getting multiplied by 1000 twice when we are creating inexpression i updated the same in description as well

2. we are using carbondatastorecreator to create store and it is using only core api so it is too difficult to do multiple loads so i added the test case for the exact scenario

chenliang613 commented:
@anubhav100  ok, got it, no need to add multiple-load for this test case.

Can you please fix this issue together in this pr , i ran PrestoAllDataTypeTest.scala, get the below error:

java.lang.NullPointerException
	at org.apache.carbondata.presto.util.CarbonDataStoreCreator$.executeGraph(CarbonDataStoreCreator.scala:498)
	at org.apache.carbondata.presto.util.CarbonDataStoreCreator$.createCarbonStore(CarbonDataStoreCreator.scala:134)
	at org.apache.carbondata.presto.integrationtest.PrestoAllDataTypeTest.beforeAll(PrestoAllDataTypeTest.scala:40)
	at org.scalatest.BeforeAndAfterAll$class.beforeAll(BeforeAndAfterAll.scala:187)
	at org.apache.carbondata.presto.integrationtest.PrestoAllDataTypeTest.beforeAll(PrestoAllDataTypeTest.scala:28)
	at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:253)
	at org.apache.carbondata.presto.integrationtest.PrestoAllDataTypeTest.run(PrestoAllDataTypeTest.scala:28)
	at org.scalatest.tools.SuiteRunner.run(SuiteRunner.scala:55)
	at org.scalatest.tools.Runner$$anonfun$doRunRunRunDaDoRunRun$3.apply(Runner.scala:2563)
	at org.scalatest.tools.Runner$$anonfun$doRunRunRunDaDoRunRun$3.apply(Runner.scala:2557)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at org.scalatest.tools.Runner$.doRunRunRunDaDoRunRun(Runner.scala:2557)
	at org.scalatest.tools.Runner$$anonfun$runOptionallyWithPassFailReporter$2.apply(Runner.scala:1044)
	at org.scalatest.tools.Runner$$anonfun$runOptionallyWithPassFailReporter$2.apply(Runner.scala:1043)
	at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:2722)
	at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:1043)
	at org.scalatest.tools.Runner$.run(Runner.scala:883)
	at org.scalatest.tools.Runner.run(Runner.scala)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.runScalaTest2(ScalaTestRunner.java:138)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.main(ScalaTestRunner.java:28)

anubhav100 commented:
@chenliang613 my pr #1695 which has the title 'Sparksql query result is not same as presto on same sql'  has already resolved this issue in latest master the reason you get error is pr #1695 get merged in between to the master branch but i did not rebased this branch with latest master since there was no merge conflict so you get the error logs but now i  rebased this branch with latest master please try running test cases once again it will work with out any error i already tested it

CarbonDataQA commented:
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2316/

CarbonDataQA commented:
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1100/

ravipesala commented:
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2547/

anubhav100 commented:
retest sdv please

chenliang613 commented:
LGTM
